tree.go

```diff
@@ -1230,41 +1259,48 @@ func (n *LeafNode) Commit() *Point {
 // fillSuffixTreePoly takes one of the two suffix tree and
 // builds the associated polynomial, to be used to compute
 // the corresponding C{1,2} commitment.
-func fillSuffixTreePoly(poly []Fr, values [][]byte) int {
+func fillSuffixTreePoly(poly []Fr, values [][]byte) (int, error) {
 	count := 0
 	for idx, val := range values {
 		if val == nil {
 			continue
 		}
 		count++
 
-		leafToComms(poly[(idx<<1)&0xFF:], val)
+		err := leafToComms(poly[(idx<<1)&0xFF:], val)
+		if err != nil {
+			return 0, err
+		}
 	}
-	return count
+	return count, nil
 }
 
 // leafToComms turns a leaf into two commitments of the suffix
 // and extension tree.
-func leafToComms(poly []Fr, val []byte) {
+func leafToComms(poly []Fr, val []byte) error {
 	if len(val) == 0 {
-		return
+		return nil
 	}
 	if len(val) > 32 {
-		panic(fmt.Sprintf("invalid leaf length %d, %v", len(val), val))
+		return fmt.Errorf("invalid leaf length %d, %v", len(val), val)
 	}
 	var (
 		valLoWithMarker [17]byte
 		loEnd           = 16
+		err             error
 	)
 	if len(val) < loEnd {
 		loEnd = len(val)
 	}
 	copy(valLoWithMarker[:loEnd], val[:loEnd])
 	valLoWithMarker[16] = 1 // 2**128
-	FromLEBytes(&poly[0], valLoWithMarker[:])
+	if err = FromLEBytes(&poly[0], valLoWithMarker[:]); err != nil {
+		return err
+	}
 	if len(val) >= 16 {
-		FromLEBytes(&poly[1], val[16:])
+		err = FromLEBytes(&poly[1], val[16:])
```

Comment:
Nit: despite this is correct, my paranoid side forces me to warn about: "if tomorrow somebody adds some extra code between lines 1301 and 1303, this can become a bug". 

So maybe simply do:
```
  ...
  if err := FromLEBytes(..., ...); err != nil { 
      return fmt.Errorf("converting to bytes: %s", err)
  }

   return nil
}
```